File: software/examples/java/ExampleThreshold.java
import com.tinkerforge.IPConnection;
import com.tinkerforge.BrickletLaserRangeFinderV2;

public class ExampleThreshold {
	private static final String HOST = "localhost";
	private static final int PORT = 4223;

	// Change XYZ to the UID of your Laser Range Finder Bricklet 2.0
	private static final String UID = "XYZ";

	// Note: To make the example code cleaner we do not handle exceptions. Exceptions
	//       you might normally want to catch are described in the documentation
	public static void main(String args[]) throws Exception {
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletLaserRangeFinderV2 lrf =
		  new BrickletLaserRangeFinderV2(UID, ipcon); // Create device object

		ipcon.connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Turn laser on and wait 250ms for very first measurement to be ready
		lrf.setEnable(true);
		Thread.sleep(250);

		// Add distance listener
		lrf.addDistanceListener(new BrickletLaserRangeFinderV2.DistanceListener() {
			public void distance(int distance) {
				System.out.println("Distance: " + distance + " cm");
			}
		});

		// Configure threshold for distance "greater than 20 cm"
		// with a debounce period of 1s (1000ms)
		lrf.setDistanceCallbackConfiguration(1000, false, '>', 20, 0);

		System.out.println("Press key to exit"); System.in.read();

		// Turn laser off
		lrf.setEnable(false);

		ipcon.disconnect();
	}
}

File: software/examples/go/example_threshold.go
package main

import (
	"fmt"
	"github.com/Tinkerforge/go-api-bindings/ipconnection"
	"github.com/Tinkerforge/go-api-bindings/laser_range_finder_v2_bricklet"
	"time"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your Laser Range Finder Bricklet 2.0.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	lrf, _ := laser_range_finder_v2_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	// Turn laser on and wait 250ms for very first measurement to be ready
	lrf.SetEnable(true)
	time.Sleep(250 * time.Millisecond)

	lrf.RegisterDistanceCallback(func(distance int16) {
		fmt.Printf("Distance: %d cm\n", distance)
	})

	// Configure threshold for distance "greater than 20 cm"
	// with a debounce period of 1s (1000ms).
	lrf.SetDistanceCallbackConfiguration(1000, false, '>', 20, 0)

	fmt.Print("Press enter to exit.")
	fmt.Scanln()

	// Turn laser off
	lrf.SetEnable(false)
}
